### DISCUSSION TITLE
How long does github forward renamed/moved repos?

### DISCUSSION THREAD

**Discussion by JayCanuck** [community/community#22669] [New to GitHub]
<p>Hi!</p>
<p>I’m in a situation where I’d like to move repos from one organization to another (and in some cases rename as well).</p>
<p>I’ve heard Github will automatically forward git requests for the old repo location to the new one (in cloning, etc.). Does anybody know how long the forwarding will happen for or is it permanent?</p>
<p>Thanks for any help anyone can provide.</p>

**Comment by mpboom** [comment]
<p>From this help page: <a href="https://help.github.com/articles/transferring-a-repository-owned-by-your-organization/:" rel="nofollow noopener">https://help.github.com/articles/transferring-a-repository-owned-by-your-organization/</a> :</p>
<pre><code>If you create a new repository under your account with the same name as the transferred repository, existing redirects to the transferred repository will break. Instead, use a different name for the new repository.
</code></pre>

**Comment by JayCanuck** [comment, marked as answer]
<p>I got a response from Github staff. May help others wondering in the future:</p>
<blockquote>
<p>If you do both a transfer and rename, the redirect will still apply.</p>
</blockquote>
<blockquote>
<p>We keep redirects for moved and renamed repos indefinitely, but new repos take priority over redirects. That means if a new repo is created at an old location, the redirects will break in favor of displaying the new repo, for as long as that new repo exists.</p>
</blockquote>
<p>Sounds like its indefinite unless a same-naned repo is created at the old location :slight_smile:</p>

**Comment by codekandis** [comment]
<p>Had the same question today.</p>
<p>While being curious I created a new repo with the old name and then I deleted it. The redirect took place again.</p>

**Reply by DuffsDevice** [reply]
Nice

**Comment by aslimacc** [comment]
<p>And how can I disable the redirection in the old location (to prevent people using the old repo ) ?</p>

**Reply by DuffsDevice** [reply]
Yeah, that's a good question

**Reply by mpboom** [reply]
Create a new repository on that URL, it will then show the new repo and thus disable any redirection.

**Comment by joshuacwnewton** [comment]
This issue mentions `git` redirects. But, I was curious about redirects for other kinds of links (like assets attached to GitHub releases).

So, I did the following test:

1. Created a fresh test repo ([repo-rename-test](https://github.com/joshuacwnewton/repo-rename-test)).
2. Created a release on the test repo and attached a release asset ([test.zip](https://github.com/joshuacwnewton/repo-rename-test/releases/download/test-release/test.zip)).
3. Renamed the repo ([repo-rename-test-RENAMED](https://github.com/joshuacwnewton/repo-rename-test-RENAMED)).
4. Tested whether the release asset is still accessible using the link with the old repo name.

Result: The asset can be accessed using links that contain both the old and new repo names! 

However, I wasn't sure if this was truly due to URL redirection, or whether it was due to GitHub's servers caching the file somehow. (?)

So, I did another quick test:

5. Modified the contents of the release asset on the newly renamed repo.
6. Tested whether the "old repo name" link will return the new, modified asset.

Result: The "old repo name" returns the updated asset. So, yes! This redirection applies not just to `git`, but to release assets as well.

**Comment by 0xturboblitz** [comment]
New response from Github Support confirming answer above:
```
The redirects will persist until either an owner of the repository asks GitHub support to remove them, or a repository with the same name is created. Even if the username is claimed, the redirects will continue as long as a repo with the same name is not created.
```
